### ISSUE TITLE
RadListView's scrollWithAmount sets incorrect position (after mandatory refresh)

### ISSUE THREAD

**Issue by abhayastudios** [telerik/nativescript-ui-feedback#780]
Hi guys!

I have a need to use a ListView, go into a detail page (save the scroll position of the ListView) and then come back to the exact position in the ListView (set the previous scroll position), so for the user it seems as if he is right back where he left off in the ListView.

I am doing this inside a `<router-outlet>` and have set a custom `RouteReuseStrategy` so that the component containing the ListView is reused.

Following the discussion [here](https://github.com/NativeScript/NativeScript/issues/2521), I am currently using a RadListView for that, where in the RadListView `(itemTap)` event, I record the position:

```
scrollOffset = this._listRef.nativeElement.getScrollOffset();
```

and then when coming back I do something like this:
```
if (isIOS) {
  this._listRef.nativeElement.refresh(); // required on iOS to redraw the ListView
}

setTimeout(() => {
  this._listRef.nativeElement.scrollWithAmount(scrollOffset, false);

  setTimeout(() => {
    console.log(`post offset: ${this._listRef.nativeElement.getScrollOffset()}`);
  },10);
},10);
```
This works fine on Android, but on iOS it behaves very funky, where it seems to be scrolling to "random" offsets in the list (and then when I scroll slowly it starts fast-forward jumping to offsets positions that are not close).

For example, if I scroll down and `getScrollOffset()` returns for example 1447.5, then I after setting it to that offset with `scrollWithAmount()` and then again request the offset it could now say that the offset is 2050.5.

When I don't call the `refresh()` then the position is working fine (then I don't even need to use `scrollWithAmount` because the RadListView is still correctly positioned due to the  RouteReuseStrategy. However, then I have another problem with is that the visible part of the RadListView then shows blank images on iOS (when I start scrolling, then the images appear). I tried to get iOS to redraw the view by doing something like below, but this does not help either:

```
this._listRef.nativeElement.ios.setNeedsDisplay();
```

Thanks for any pointers!

P.S.

I am using NS 4.2 and v3.5.11 of the `nativescript-ui-listview` plugin.

**Comment by abhayastudios** [comment]
OK after some more playing around I think I found out what is going on with `scrollWithAmount`. Apparently it does not handle the scaling factor properly:

```
// will output for example 2442.5
let scrollOffset = this._listRef.nativeElement.getScrollOffset();
console.log(`pre scrollOffset: ${scrollOffset}`);

// set to the previous position
this._listRef.nativeElement.scrollWithAmount(scrollOffset, false);

// now position is set to 4885 (on a scale factor 2 screen)
console.log(`post offset: ${this._listRef.nativeElement.getScrollOffset()}`);
```

I think this is a bug and that `scrollWithAmount` should handle this properly by itself.

Thanks!

**Comment by abhayastudios** [comment]
Another observation that may be a clue as to what is the cause of this weird behavior is that apparently there is some relation to the screen scale factor.

If I comment out the refresh (causing the images in the "active view" to be blank") like below, then the 
post offset output will always be double of the offset that I fed to `scrollWithAmount`, which is exactly the screen scale of the device I am testing on.
 
```
if (isIOS) {
  //this._listRef.nativeElement.refresh(); // required on iOS to redraw the ListView
}

setTimeout(() => {
  this._listRef.nativeElement.scrollWithAmount(scrollOffset, false);

  setTimeout(() => {
    console.log(`post offset: ${this._listRef.nativeElement.getScrollOffset()}`);
  },10);
},10);
```

Thanks!

**Comment by zbranzov** [comment]
@abhayastudios Unfortunately, there is an issue when using tap event in the RadListView template. If you slightly scroll an item it will be clicked on iOS which most probably sends wrong offset.
I tested with and without the refresh() and both doesn't work well:
```
if (isIOS) {
  //this._listRef.nativeElement.refresh(); // required on iOS to redraw the ListView
}
```
So the solution I came up with is to use (itemTap)="navigateToDetails($event)" of the ListView and use scrollToIndex instead scrollWithAmount.
[Here is a link to sample project](https://github.com/zbranzov/ns-angular-nested-router). Take a look at the dogs tab.

**Comment by zbranzov** [comment]
Related to https://github.com/telerik/nativescript-ui-feedback/issues/1048

**Comment by tbozhikov** [comment]
Maybe related to #1055

**Comment by tbozhikov** [comment]
As #1055 is now released (RLV v6.1.0), we need to test if this issue still persists.

**Comment by tsonevn** [comment]
I was unable to recreate the issue with latest nativescript-ui-listview (v.6.2.0). I am closing the issue for now. If the problem appears again, I will reopen it.
